src/components/Highlighter/StringHighlighter/__tests__/StringHighlighter.spec.tsx

```diff
@@ -1,83 +1,100 @@
 import React from 'react';
-import toJson from 'enzyme-to-json';
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
 import { StringHighlighter } from '../StringHighlighter';
 
 describe('StringHighlighter', () => {
+    let view;
     const string = 'We are looking for php, java and javascript developer';
     const searchTerms = [];
-    const ignoreDiacritics = true;
-    const ignoreCase = true;
-    // eslint-disable-next-line react/display-name, react/prop-types
     const highlightRenderer = ({ substring, ...props }) => <b {...props}>{substring}</b>;
-    // eslint-disable-next-line react/display-name, react/prop-types
-    const renderComponent = (props) => {
-        return mount(
-            <StringHighlighter
-                string={string}
-                searchTerms={searchTerms}
-                highlighterCoreOptions={(ignoreDiacritics, ignoreCase)}
-                highlightRenderer={highlightRenderer}
-                {...props}
-            />
-        );
+    const defaultProps = {
+        string,
+        searchTerms,
+        highlightRenderer,
     };
+    const rerenderView = (props) => {
+        view.rerender(<StringHighlighter {...defaultProps} {...props} />);
+    };
+    beforeEach(() => {
+        view = render(<StringHighlighter {...defaultProps} />);
+    });
 
     describe('not highlighted', () => {
         it('should initially render empty component correctly', () => {
-            const wrapper = renderComponent({
+            rerenderView({
                 string: '',
             });
-            expect(toJson(wrapper)).toMatchSnapshot();
+
+            expect(view.container).toMatchSnapshot();
         });
+
         it('should not highlight not existed terms', () => {
-            const wrapper = renderComponent({
+            rerenderView({
                 searchTerms: ['notExisting', 'dev', 'a'],
             });
-            expect(toJson(wrapper)).toMatchSnapshot();
+
+            expect(view.container).toMatchSnapshot();
+            expect(screen.getByText(string)).toBeInTheDocument();
```

Comment:
I really like that you added specific tests like that, I think these make them a lot more resilient. Thanks!
However, it is a bit hidden logic why we are looking for the string and not for the highlights. Since the highlighter tag is only in the tests here, maybe it is ok to use getByTestId, and add the test id to the `highlightRenderer` at the top. Then here and in the tests below you can check how many of those are present. What do you think?

Reply:
:+1:

Reply:
nicely done. thanks